services/management/grpc/actions_server.go

```diff
@@ -260,38 +274,62 @@ func (s *actionsServer) StartMongoDBExplainAction(ctx context.Context, req *mana
 	}, nil
 }
 
-// StartPTSummaryAction starts pt-summary action.
+// StartPTSummaryAction starts pt-summary action. If the time since the last successful start of pt-summary action is lower than ptSummaryRefreshPeriod,
+// the response from the last action will be used. If the time is longer, the new pt-summary action will be called.
 //nolint:lll
-func (s *actionsServer) StartPTSummaryAction(ctx context.Context, req *managementpb.StartPTSummaryActionRequest) (*managementpb.StartPTSummaryActionResponse, error) {
-	agents, err := models.FindPMMAgentsRunningOnNode(s.db.Querier, req.NodeId)
+func (s *actionsServer) StartPTSummaryAction(ctx context.Context, psReq *managementpb.StartPTSummaryActionRequest) (*managementpb.StartPTSummaryActionResponse, error) {
+	// Gets current timestamp
+	timeNow := time.Now().Unix()
+
+	// Gets pointers to agents running on the node
+	psAgents, err := models.FindPMMAgentsRunningOnNode(s.db.Querier, psReq.NodeId)
 	if err != nil {
 		return nil, status.Errorf(codes.NotFound, "No pmm-agent running on this node")
 	}
 
-	agents = models.FindPMMAgentsForVersion(s.l, agents, pmmAgent2100)
-	if len(agents) == 0 {
+	// Filter by the version
+	psAgents = models.FindPMMAgentsForVersion(s.l, psAgents, pmmAgent2100)
```

Comment:
That comment is right there, in the function name

Reply:
I had to open up the function to understand it. So instead of 1 sec I spent 30. That's why I put there the comment to reduce the investigation time of the reader. Time is expensive.

Reply:
Then we probably should find a better name for the function to be able to understand what it does from the name. Otherwise, there is high chance that people will read comments and the code in question

Reply:
Also, it is just a question of familiarity with the code base. We can safely assume that all Go developers know what `time.Now().Unix()` does without checking the source code, right? We assume the same about most of the PMM code from PMM developers after some time.

Then again, I don't imply that we should not add comments at all. It is just that one is not helpful there.

Reply:
Yes. An here starts the process: Requirement - design - discussion - approval - implementation - documentation.

Reply:
Not sure I understand what you mean there.

Reply:
Simply said: Decide what what we want to do before we cut the patient. I feel here it is opposite. First we cut.
I was referring to process development cycle. And it would also be good to have a company coding standard document (coding rules). No judge can work without the law.

Reply:
But we _do_ have it. Please check Slack